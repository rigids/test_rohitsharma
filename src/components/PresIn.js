import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Form, FormGroup, FormControl, ControlLabel, Button } from 'react-bootstrap';
import { connect } from 'react-redux';
import { addItem } from '../actions/';
import { Link,BrowserRouter } from 'react-router-dom';

const multiselectList = ["One", "Two", "Three", "Four", "Five"];

class PresIn extends Component {
  constructor() {
    super();
    this.state = {
      stored_value: '',
      show_msg: false,
      stored_selection: []
      
    }
  }
  save(){
    this.props.addItem(this.state);
    // clear form
    ReactDOM.findDOMNode(this.refs.text_bar).value    =  "";
    ReactDOM.findDOMNode(this.refs.select_bar).value  =  "";
    this.setState({showMsg: true})
    // show success message for 2 seconds
    setTimeout(() => this.setState({showMsg: false}), 2000);
    
  }
  render(){
    
    const { showMsg } = this.state;
    return (
      <div className="block__wrapper">
        <div className="menu__wrapper">
          <ul>
            <li className="shadow-alt">
              <Link to='/contout'>Saved Items</Link>
            </li>
          </ul>
        </div>
        <div className="form__container shadow">
          {  
            showMsg ?
            <div className="alert alert-success">Item Saved</div>:
            <div className="alert">&nbsp;</div>
          }
          <Form>
            <FormGroup>
              <ControlLabel>Description</ControlLabel>
              <FormControl ref="text_bar"
                  componentClass="textarea" 
                  placeholder="Enter Description"
                  rows="10"
                  onChange={event => this.setState({ stored_value: event.target.value })}
                />
            </FormGroup>
            <FormGroup controlId="formControlsSelectMultiple" >
              <ControlLabel>Select</ControlLabel>
              {' '}
              <FormControl multiple componentClass="select"  ref="select_bar"
                onChange={event=> this.setState({stored_selection:[...event.target.options].filter(({selected}) => selected).map(({value}) => value)})} 
              >
                <option key="_blank" value="">Select Option</option>
                {
                  multiselectList.map((item,i) => <option key={i} value={item}>{item}</option>)
                }
              </FormControl>
            </FormGroup>
            <div className="form-button">
             
              <Button disabled={showMsg} onClick={ ()=>this.save() }>Submit</Button>
            </div>
          </Form>
        </div>
      </div>
    )
  }
}

export default connect(null,{addItem})(PresIn);