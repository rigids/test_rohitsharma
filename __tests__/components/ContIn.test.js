import React, {Connect} from 'react';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import PresIn from '../../src/components/PresIn';
import { BrowserRouter } from 'react-router-dom';

const mockStore = configureStore();

const initialState = [];

describe('Items reducer', () => {
  let wrapper;
  beforeEach(() => {
    const store = mockStore(initialState);
    wrapper = mount(
        <Provider store={store}>
          <BrowserRouter>
            <PresIn />
          </BrowserRouter>
        </Provider>
      );
  });
  
  it('Render PresIn Component with button.', () => {
      expect(wrapper.find('.btn').length).toBe(1);
  });
  
  it('Snapshot render', () => {
    expect(toJson(wrapper.render())).toMatchSnapshot();
  });
    
});


