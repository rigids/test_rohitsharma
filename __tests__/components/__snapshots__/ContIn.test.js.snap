// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items reducer Snapshot render 1`] = `
<div
  class="block__wrapper"
>
  <div
    class="menu__wrapper"
  >
    <ul>
      <li
        class="shadow-alt"
      >
        <a
          href="/contout"
        >
          Saved Items
        </a>
      </li>
    </ul>
  </div>
  <div
    class="form__container shadow"
  >
    <div
      class="alert"
    >
      Â 
    </div>
    <form
      class=""
    >
      <div
        class="form-group"
      >
        <label
          class="control-label"
        >
          Description
        </label>
        <textarea
          class="form-control"
          placeholder="Enter Description"
          rows="10"
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="control-label"
          for="formControlsSelectMultiple"
        >
          Select
        </label>
         
        <select
          class="form-control"
          id="formControlsSelectMultiple"
          multiple=""
        >
          <option
            value=""
          >
            Select Option
          </option>
          <option
            value="One"
          >
            One
          </option>
          <option
            value="Two"
          >
            Two
          </option>
          <option
            value="Three"
          >
            Three
          </option>
          <option
            value="Four"
          >
            Four
          </option>
          <option
            value="Five"
          >
            Five
          </option>
        </select>
      </div>
      <div
        class="form-button"
      >
        <button
          class="btn btn-default"
          type="button"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
